fv3_cap.F90

```diff
@@ -1250,25 +1250,32 @@ subroutine fv3_checkimport(gcomp, rc)
         call ESMF_FieldGet(fieldList(n), name=fldname, rc=rc)
         if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, line=__LINE__, file=__FILE__)) return
 
-        nf = queryImportFields(fldname)
-        timeCheck1 = NUOPC_IsAtTime(fieldList(n), invalidTime, rc=rc)
+        ! check if import field carries a valid timestamp
+        call NUOPC_GetTimestamp(fieldList(n), isValid=isValid, rc=rc)
         if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, line=__LINE__, file=__FILE__)) return
 
-        if (timeCheck1) then
-          importFieldsValid(nf) = .false.
-!         if(mtype==0) print *,'in fv3_checkimport,',trim(fldname),' is set unvalid, nf=',nf,' at time',date(1:6)
-        else
-          timeCheck2 = NUOPC_IsAtTime(fieldList(n), currTime, rc=rc)
+        if (isValid) then
+          ! if timestamp is set, check if it is valid
+          isValid = .not.NUOPC_IsAtTime(fieldList(n), invalidTime, rc=rc)
           if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, line=__LINE__, file=__FILE__)) return
+        end if
+
+        ! store field status in internal array
+        nf = queryImportFields(fldname)
+        importFieldsValid(nf) = isValid
 
-          if (.not.timeCheck2) then
-            !TODO: introduce and use INCOMPATIBILITY return codes!!!!
+        if (isValid) then
+          ! check if field is current
+          isValid = NUOPC_IsAtTime(fieldList(n), currTime, rc=rc)
+          if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, line=__LINE__, file=__FILE__)) return
+          if (.not.isValid) then
             call ESMF_LogSetError(ESMF_RC_ARG_BAD, &
-                                  msg="NUOPC INCOMPATIBILITY DETECTED: Import Field not at current time", &
+                                  msg="NUOPC INCOMPATIBILITY DETECTED: Import Field " &
+                                      // trim(fldname) // " not at current time", &
```

Comment:
Should we have: importFieldsValid(nf) = .false. here too? Or maybe we are now allowing import fields with different time stamp (not at currTime)?

Reply:
We could set `importFieldsValid(nf) = .false.` here as well if we are not planning to abort when a time incompatibility is detected. Currently, we return an ESMF error code that will cause NUOPC to abort.